src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -354,11 +355,11 @@ export const QueryEditorUI: React.FC<Props> = (props) => {
 
       editor.addCommand(monaco.KeyCode.Enter, () => {
         const newQuery = {
-          ...query,
+          ...queryRef.current,
```

Comment:
Regarding this change, I see that earlier in the file, `queryRef` exists for monaco 'commands' to access the latest query state. 
If `query` was a stale query state, why wasn't the single line editor failing a test that would update the query? 
If there is a test for an updated query state, and `query` has the updated query state, do we need queryRef at all? It only seems to be used here.